code/game/machinery/magnet.dm

```diff
@@ -361,8 +360,7 @@
 		pathpos++ // increase iterator
 
 		// Broadcast the signal
-		spawn()
-			radio_connection.post_signal(src, signal, filter = RADIO_MAGNETS)
+		INVOKE_ASYNC(radio_connection, /datum/radio_frequency/radio_connection.proc/post_signal, src, signal, filter = RADIO_MAGNETS)
 
 		if(speed == 10)
 			sleep(1)

```

Comment:
Missed a spot. this sleep is just a delay for the looping = 0 bit below, there is a timer system for setting vars you could use for that instead, avoiding the sleep, or you could make that be a world time that is checked, avoiding the need to use timers or sleeps entirely